dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -13,21 +13,16 @@ def update_last_confirmation_date
   def update
     return unless school_info_params[:school_id].present? || school_info_params[:country].present?
 
-    new_school_info_params =
-      if school_info_params[:full_address]&.blank?
-        school_info_params.except(:full_address)
-      else
-        school_info_params
-      end
+    @school_info_params.delete(:full_address) if @school_info_params[:full_address]&.blank?
```

Comment:
Now that you've got this working, there's a small bit of cleanup I'd do: It's not obvious where `@school_info_params` comes from (it's set as a side-effect of calling `school_info_params`, above.  Two solutions come to mind.

1. Use `school_info_params` instead of `@school_info_params` throughout `def update`.
   Now that you've [memoized](https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/) the `school_info_params` function, you can mutate the hash it returns and every call to that function will include your latest changes.

2. De-memoize the helper and go back to a temporary variable
   Setting `new_school_info_params = school_info_params` and the mutating the temporary variable would be fine too, and maybe a little more obvious about what's going on.

I'm leaning toward 1, despite it being a bit on the "clever" side, because it will read really nicely.

Reply:
Thank you!

Reply:
@islemaster Is it good practice to use memoization?  I am curious about which scenarios it is more beneficial to use.

Reply:
We typically memoize to avoid doing something expensive more than once.  For example, this function uses memoization to only do a database query the first time it's called:

https://github.com/code-dot-org/code-dot-org/blob/82792e36f388674d49c8ecf477ae4d614c377030/pegasus/router.rb#L284-L287

Yours is a less typical case. We're not memoizing here because filtering the params is expensive, but more because conceptually we shouldn't need to do it more than once; and because they're effectively global to the controller, and we want the ability to modify them.

So I'm not sure this is a best practice, but its enough of a well-known pattern that I don't worry it will confuse anyone.